rclrs/src/node/builder.rs

```diff
@@ -187,4 +269,59 @@ impl NodeBuilder {
             subscriptions: std::vec![],
         })
     }
+
+    /// Creates node options.
+    /// options are overrided by field values of builder.    
+    fn create_node_options(&self) -> Result<rcl_node_options_t, RclrsError> {
+        // SAFETY: No preconditions for this function.
+        let mut node_options = unsafe { rcl_node_get_default_options() };
+
+        let cstring_args = self
+            .arguments
+            .iter()
+            .map(|s| CString::new(s.as_str()).unwrap())
+            .collect::<Vec<_>>();
+        let cstring_arg_ptrs = cstring_args.iter().map(|s| s.as_ptr()).collect::<Vec<_>>();
+        // SAFETY: Getting a zero-initialized value is always safe.
+        unsafe {
+            // SAFETY: This function does not store the ephemeral cstring_args_ptrs
+            // pointers. We are passing in a zero-initialized arguments struct as expected.
+            rcl_parse_arguments(
+                cstring_arg_ptrs.len() as i32,
+                cstring_arg_ptrs.as_ptr(),
+                rcutils_get_default_allocator(),
+                &mut node_options.arguments,
+            )
+        }
+        .ok()?;
+
+        node_options.use_global_arguments = self.use_global_arguments;
+        node_options.enable_rosout = self.enable_rosout;
+        // SAFETY: No preconditions for this function.
+        node_options.allocator = unsafe { rcutils_get_default_allocator() };
+
+        Ok(node_options)
+    }
+}
+
+impl Drop for rcl_arguments_t {
```

Comment:
I think it would be slightly cleaner to just remove this impl since it will always be freed as part of `rcl_node_options_fini`. But if you disagree, feel free to leave it in.

Reply:
I can't think of an example where `rcl_arguments_t` would be used directly and not in combination with `rcl_node_options_t`, except for parsing command line arguments, but we'd use Rust's native facilities in that case. I can understand for the sake of completeness, it's up to you @Soya-Onishi if you want to keep this code, I'm leaning towards removing it.

Reply:
@esteve Thanks for your advice! I don't know whether it is related to what @nnmm says or not, but should I put statements about `rcl_arguments_t` is finished in `rcl_node_options_fini()`?

Reply:
Yes, we're both saying the same :) You could add a comment `// This also finalizes the rcl_arguments contained in the node options`  to the `Drop for rcl_node_options_t`, and then just delete the `Drop for rcl_arguments_t`.